src/main/java/hudson/plugins/throttleconcurrents/ThrottleJobProperty.java

```diff
@@ -229,29 +243,82 @@ public boolean isThrottleMatrixConfigurations() {
         assert category != null && !category.equals("");
         List<Queue.Task> categoryTasks = new ArrayList<Queue.Task>();
         Collection<ThrottleJobProperty> properties;
-        DescriptorImpl descriptor = Jenkins.getActiveInstance().getDescriptorByType(DescriptorImpl.class);    
+        DescriptorImpl descriptor = Jenkins.getActiveInstance().getDescriptorByType(DescriptorImpl.class);
         synchronized (descriptor.propertiesByCategoryLock) {
-            Map<ThrottleJobProperty,Void> _properties = descriptor.propertiesByCategory.get(category);
+            Map<ThrottleJobProperty, Void> _properties = descriptor.propertiesByCategory.get(category);
             properties = _properties != null ? new ArrayList<ThrottleJobProperty>(_properties.keySet()) : Collections.<ThrottleJobProperty>emptySet();
         }
         for (ThrottleJobProperty t : properties) {
             if (t.getThrottleEnabled()) {
                 if (t.getCategories() != null && t.getCategories().contains(category)) {
-                    Job<?,?> p = t.owner;
+                    Job<?, ?> p = t.owner;
                     if (/*is a task*/ p instanceof Queue.Task && /* not deleted */getItem(p.getParent(), p.getName()) == p &&
                         /* has not since been reconfigured */ p.getProperty(ThrottleJobProperty.class) == t) {
                         categoryTasks.add((Queue.Task) p);
                         if (p instanceof MatrixProject && t.isThrottleMatrixConfigurations()) {
-                            for (MatrixConfiguration mc : ((MatrixProject)p).getActiveConfigurations()) {
+                            for (MatrixConfiguration mc : ((MatrixProject) p).getActiveConfigurations()) {
                                 categoryTasks.add(mc);
                             }
                         }
                     }
                 }
             }
         }
+
         return categoryTasks;
     }
+
+    static List<ThrottledStepInfo> getThrottledPipelinesForCategory(String category) {
+        List<ThrottledStepInfo> throttledPipelines = new ArrayList<>();
+
+        DescriptorImpl descriptor = Jenkins.getActiveInstance().getDescriptorByType(DescriptorImpl.class);
+        for (Map.Entry<String,Integer> currentPipeline : descriptor.getThrottledPipelinesForCategory(category).entrySet()) {
+            Run<?,?> flowNodeRun = Run.fromExternalizableId(currentPipeline.getKey());
+
+            if (flowNodeRun == null) {
+                // No run found, so remove the throttle.
+                descriptor.removeThrottledPipelineForCategory(currentPipeline.getKey(), category, null);
+            } else if (!(flowNodeRun instanceof FlowExecutionOwner.Executable)) {
+                // If for some reason we've somehow ended up with a non-pipeline job, remove the throttle.
```

Comment:
Is there an easy way to go from some easily serializable identifier for `FlowExecutionOwner` to an actual instance of `FlowExecutionOwner`, like with `Run.fromExternalizableId`?